storage/libradb/src/schema/transaction_by_account/mod.rs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module defines physical storage schema for a transaction index via which the version of a
+//! transaction sent by `account_address` with `sequence_number` can be found. With the version one
+//! can resort to `SignedTransactionSchema` for the transaction content.
+//!
+//! ```text
+//! |<-------key------->|<-value->|
+//! | address | seq_num | txn_ver |
+//! ```
+
+use crate::schema::{ensure_slice_len_eq, TRANSACTION_BY_ACCOUNT_CF_NAME};
+use byteorder::{BigEndian, ReadBytesExt, WriteBytesExt};
+use failure::prelude::*;
+use schemadb::{
+    define_schema,
+    schema::{KeyCodec, ValueCodec},
+};
+use std::{convert::TryFrom, mem::size_of};
+use types::{
+    account_address::{AccountAddress, ADDRESS_LENGTH},
+    transaction::Version,
+};
+
+define_schema!(
+    TransactionByAccountSchema,
+    Key,
+    Value,
+    TRANSACTION_BY_ACCOUNT_CF_NAME
+);
+
+type SeqNum = u64;
+type Key = (AccountAddress, SeqNum);
+
+type Value = Version;
+
+impl KeyCodec<TransactionByAccountSchema> for Key {
+    fn encode_key(&self) -> Result<Vec<u8>> {
+        let (ref account_address, seq_num) = *self;
+
+        let mut encoded = account_address.to_vec();
+        encoded.write_u64::<BigEndian>(seq_num)?;
+
+        Ok(encoded)
+    }
+
+    fn decode_key(data: &[u8]) -> Result<Self> {
+        ensure_slice_len_eq(data, size_of::<Self>())?;
+
+        let event_key = AccountAddress::try_from(&data[..ADDRESS_LENGTH])?;
+        let seq_num = (&data[ADDRESS_LENGTH..]).read_u64::<BigEndian>()?;
+
+        Ok((event_key, seq_num))
+    }
+}
+
+impl ValueCodec<TransactionByAccountSchema> for Version {
```

Comment:
Do you want to attach this implementation to `Version`?  It seems `Value` is more appropriate.

Reply:
it's in effect only when you use `TransactionByAccountSchema` to the call site module. I think it's ergonomic in rust world?

I should remove `Value` btw.